pkg/module/common/module.go

```diff
@@ -76,14 +75,15 @@ func (m *Module) ContainsExpectedOwnerReference(ownerName string) bool {
 	return false
 }
 
-func NewFromModule(module *Module) *manifestV1alpha1.Manifest {
-	fromServer := manifestV1alpha1.Manifest{}
-	fromServer.SetGroupVersionKind(module.GroupVersionKind())
-	fromServer.SetNamespace(module.GetNamespace())
-	fromServer.SetName(module.GetName())
-	return &fromServer
-}
+const maxModuleNameLength = 253
 
-func CreateModuleName(moduleName, kymaName string) string {
-	return fmt.Sprintf("%s-%s", kymaName, moduleName)
+func CreateModuleName(fqdn, kymaName string) string {
+	hash := fnv.New32()
+	_, _ = hash.Write([]byte(fqdn))
+	hashedFQDN := hash.Sum32()
+	name := fmt.Sprintf("%s-%v", kymaName, hashedFQDN)
```

Comment:
I actually have some concern here for the naming, if we hash the module name, then by list the manifest CR with kubectl, basically don't know which module it is by first look, maybe get complained later for troubleshooting purpose.

Reply:
I changed the manifests that are generated to have also an annotation for the module name. I would like to have used a label but unfortunately the FQDN is not a RFC compliant label.

All modules can still be received by introspecting the kyma CR.